types/aws-lambda/index.d.ts

```diff
@@ -501,6 +501,52 @@ export interface AuthResponseContext {
     [name: string]: any;
 }
 
+/**
+ * CodePipeline events
+ * https://docs.aws.amazon.com/codepipeline/latest/userguide/actions-invoke-lambda-function.html
+ */
+export interface S3ArtifactLocation {
+    bucketName: string;
+    objectKey: string;
+}
+export interface S3ArtifactStore {
+    type: 'S3';
+    s3Location: S3ArtifactLocation;
+}
+
+export type ArtifactLocation = S3ArtifactStore;
```

Comment:
Presumably this would become a union of other artifact location types? Just checked [the docs](https://docs.aws.amazon.com/codepipeline/latest/APIReference/API_ArtifactLocation.html), and it does only have the `S3` type, weird. As is it's a little confusing, could maybe use a comment for whoever follows up.